lib/ansible/executor/task_executor.py

```diff
@@ -578,6 +578,14 @@ def _execute(self, variables=None):
                 # pc compare, left here for old plugins, but should be irrelevant for those
                 # using get_option, since they are cleared each iteration.
                 self._play_context.remote_addr != self._connection._play_context.remote_addr):
+            # attempt to close existing connection if any
+            if self._connection:
+                try:
+                    self._connection.close()
+                except AttributeError:
+                    pass
+                except Exception as e:
+                    display.debug(u"error closing connection: %s" % to_text(e))
```

Comment:
aside from missing tests and a changelog ... this might not be a good approach as other connections are reused (even if not for the current task) in the play and this would require the costly re-initialization of them.

I know the winrm connection does somethings differently but a general reaper at the end of the run might be a better solution, though many connections might accumulate during the run.  Another issue is that some setups reuse the connections created by Ansible (mostly ssh) , so it would have to be a toggle or a per connection property.

Reply:
Thanks for valuable comment, I made this PR more like hotfix to get some guidance how to fix it properly - I am sorry if it should been discussion in Issue first... It seemed to me that task_executor always closes connection at the end of `.run()` anyway and wasn't aware that connection can be retained over `self._connection.close()` call there.

I think general reaper at the end might not be good option either as default WinRM quota is quite low (I think its 30 connections per user) so if playbook has a task with more than 30 loop items it will blew before reaper kicks in...